### PR TITLE
implemented auto-generating and editable slugs

### PR DESCRIPTION
**Pull request by Sajal0208**
## What does this PR do?

This PR resolves the issue mentioned in https://github.com/SuperteamDAO/earn/issues/479

Open to any suggestions which might improve the UI or the UX of the Form.

### CODE REVIEW

**Review by a20hek — CHANGES_REQUESTED** [review]
would like the implementation to have 
1. the initial slug should not be a random slug, but a slug code similar to https://github.com/SuperteamDAO/earn/blob/main/src/pages/api/bounties/create.ts#L9-L44
2. slug should change on title input change, but again – not a random change, it should fetch and check what the appropriate slug should be for that particular title, and then load that particular slug

File: src/features/sponsor-dashboard/components/Submissions/SubmissionDetails.tsx [hunk 1]
```diff
@@ -171,7 +172,7 @@ export const SubmissionDetails = ({
   }) => {
     setIsPaying(true);
     try {
-      let transaction;
+      const transaction = new Transaction();
```
Comment (a20hek) [inline src/features/sponsor-dashboard/components/Submissions/SubmissionDetails.tsx:175]: my changes from a different PR are showing up in this PR's diff
so i would suggest you to open a separate PR by checking out from the latest staging branch

File: src/pages/api/listings/slug.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import slugify from 'slugify';
+
+import { prisma } from '@/prisma';
+
+const checkSlug = async (slug: string): Promise<boolean> => {
+  try {
+    const bounty = await prisma.bounties.findFirst({
+      where: {
+        slug,
+        isActive: true,
```
Comment (a20hek) [inline src/pages/api/listings/slug.ts:11]: remove `isActive: true`

**Review by a20hek — CHANGES_REQUESTED** [review]
please perform the requested changes, and also fix naming of functions 
rest looks good to me

**Comment by Sajal0208** [comment]
@a20hek done with all the changes. 
opened a new PR: #516 
thankyou :)

**Comment by a20hek** [comment]
> @a20hek done with all the changes. opened a new PR: #516 thankyou :)

cool, thanks
hence closing this PR

**Pull request closed without merging**